main.go

```diff
@@ -370,7 +373,37 @@ func main() {
 		}
 	}
 	// handle profiler sink close on exit
-	defer prfl.Stop()
+	defer func() {
+		logger.Infow("closing profiler sink client before exiting", "sink", prfl.GetSinkName())
+		prfl.Stop()
+	}()
+
+	// tracer sink
+	ts, err := tracer.GetSink(logger, tracertypes.SinkDriver(cfg.Controller.TracerSink))
+	if err != nil {
+		logger.Errorw("error while creating profiler sink, switching to noop", "error", err)
+
+		ts, err = tracer.GetSink(logger, tracertypes.SinkDriverNoop)
+
+		if err != nil {
+			logger.Errorw("error while creating noop profiler sink", "error", err)
+		}
+	}
+	// handle tracer sink close on exit
+	defer func() {
+		logger.Infow("closing tracer sink client before exiting", "sink", ts.GetSinkName())
+
+		if err := ts.Stop(); err != nil {
+			logger.Errorw("error closing tracer sink client", "sink", ms.GetSinkName(), "error", err)
+		}
+	}()
```

Comment:
Not sure it makes sense to do this for a Noop sink?

Reply:
It doesn't; however it's useful for the DD sink and potential others.